### PR TITLE
fix(TDP-5968): Migrate to dataset name in preparation

### PR DESCRIPTION
**Pull request by fhuaulme**
[POJO changes]
* Add dataSetName in Preparation POJO.
* Deprecate dataSetId.
* Adapt DTOs to include dataset name.

[Data set name update]
* If dataset name misses in list operation, performs a look up to dataset service.
* Create a preparation also add linked dataset name.
* Add dataset name update operation in preparation listener for kafka event.

**Link to the JIRA issue**
https://jira.talendforge.org/browse/TDP-5968

**Please check if the PR fulfills these requirements**
- [x] The commit(s) message(s) follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md#commit-message-format)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] The code coverage on new code is > 75 % for backend and > 95% for frontend
- [ ] The new code does not introduce new technical issues (sonar / eslint)
- [x] Functional tests have been performed
- [ ] Docker configuration files for config-std or config-cloud profiles are impacted

**Please check the browsers you've tested on**
- [ ] Chrome, Firefox, Safari, Edge, IE11
- [ ] No, that's bad, this PR should not be merged !
- [x] No, and no need to (backend changes only)

### CODE REVIEW

File: dataprep-backend-service/src/main/java/org/talend/dataprep/conversions/inject/DataSetNameInjection.java [hunk 1]
```diff
@@ -5,25 +5,21 @@
...
 import org.talend.dataprep.exception.TDPException;
 import org.talend.dataprep.exception.error.CommonErrorCodes;
 import org.talend.dataprep.security.Security;
+import org.talend.dataprep.security.SecurityProxy;
 
 import com.google.common.cache.Cache;
 import com.google.common.cache.CacheBuilder;
-import org.talend.dataprep.security.SecurityProxy;
 
 public class DataSetNameInjection implements BiFunction<PreparationDTO, PreparationListItemDTO, PreparationListItemDTO> {
```
[the remark this answers is not in the document]
Comment (fhuaulme) [reply dataprep-backend-service/src/main/java/org/talend/dataprep/conversions/inject/DataSetNameInjection.java]: 👎 
Yes, it useful in the allegedly unlikely case were the migration that set dataset name in preparation did not run. The code is aimed to work without the execution of the migration.

File: dataprep-backend-service/src/main/java/org/talend/dataprep/conversions/inject/DataSetNameInjection.java [hunk 2]
```diff
@@ -43,21 +39,26 @@ public DataSetNameInjection() {
...
+                if (tenantCache.getIfPresent(dto.getDataSetId()) == null) {
+                    securityProxy.asTechnicalUserForDataSet();
+                    try {
+                        tenantCache.put(dto.getDataSetId(), datasetClient.getDataSetMetadata(dto.getDataSetId()).getName());
+                    } finally {
+                        securityProxy.releaseIdentity();
+                    }
                 }
+                item.getDataSet().setDataSetName(tenantCache.getIfPresent(dto.getDataSetId()));
+                return item;
```
[the remark this answers is not in the document]
Comment (fhuaulme) [reply dataprep-backend-service/src/main/java/org/talend/dataprep/conversions/inject/DataSetNameInjection.java]: 👍

File: dataprep-backend-service/src/main/java/org/talend/dataprep/preparation/store/PersistentPreparation.java [hunk 3]
```diff
@@ -36,9 +36,15 @@
     @Version
     private Long version;
 
-    /** The dataset id. */
+    /**
+     * The dataset id.
+     * @deprecated Use {@link #dataSetName} instead.
+     */
+    @Deprecated
     private String dataSetId;
```
[the remark this answers is not in the document]
Comment (fhuaulme) [reply dataprep-backend-service/src/main/java/org/talend/dataprep/preparation/store/PersistentPreparation.java]: I removed the useless call to the DataSetGetMetadata command. The rest is indeed out of scope of current PR (and very error prone).

File: dataprep-preparation/src/main/java/org/talend/dataprep/preparation/event/PreparationEventUtil.java [hunk 4]
```diff
@@ -109,6 +115,9 @@ private void removePreparationStepRowMetadata(String dataSetId) {
             preparationRepository
                     .list(PersistentPreparation.class, eq("dataSetId", dataSetId)) //
                     .forEach(preparation -> {
+                        // Update dataset name in preparation
```
[the remark this answers is not in the document]
Comment (fhuaulme) [reply dataprep-preparation/src/main/java/org/talend/dataprep/preparation/event/PreparationEventUtil.java]: 👍

File: dataprep-backend-service/src/main/java/org/talend/dataprep/preparation/store/PersistentPreparation.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (aleboulanger) [reply dataprep-backend-service/src/main/java/org/talend/dataprep/preparation/store/PersistentPreparation.java]: ok i will create a work item for this subject

**Pull request merged**